packages/space-opera/src/components/materials_panel/materials_panel.ts

```diff
@@ -393,15 +394,27 @@ export class MaterialPanel extends ConnectedLitElement {
     }
   }
 
-  async onTextureUpload(uri: string, textureInfo: TextureInfo) {
-    if (this.thumbnailsById.has(uri)) {
+  async onTextureUpload(
+      detail, texturePicker: TexturePicker, textureInfo: TextureInfo) {
+    const {url, type} = detail;
+    if (this.thumbnailsById.has(url)) {
       console.log('URL collision! Texture not updated.');
       return;
     }
-    const texture = await getModelViewer()?.createTexture(uri);
+    const texture = await getModelViewer()?.createTexture(url);
     if (texture == null) {
       return;
     }
+    // This hack is because GLTFExporter checks if format is RGB vs RGBA to
+    // decide if it should save as JPEG vs PNG. However, TextureLoader sets
+    // format based on if the url ends in .jpg, which does not work for an
+    // ObjectURL like we're passing here. So, to keep from inflating all JPEGs
+    // to PNGs, we do this hack, and since I don't have a separate three.js dep
+    // in here, I'm setting RGBFormat by number instead of name.
+    if (type === 'image/jpeg') {
+      texture.source[$threeTexture].format = 1022;  // RGBFormat
```

Comment:
@mrdoob Please take a look at this comment. This sort of feels like a bug in `TextureLoader`, but I'm not really sure how it could be fixed?